atlasdb-docker-test-utils/src/main/java/com/palantir/atlasdb/testing/DockerProxySelector.java

```diff
@@ -58,6 +58,6 @@ public DockerProxySelector(Cluster containers, Supplier<ProjectInfoMappings> pro
 
     @Override
     public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
-        throw Throwables.propagate(ioe);
+        Preconditions.checkArgument(uri != null && sa != null && ioe != null, "Invalid connectFailed call");
```

Comment:
checkNotNull won't return `IllegalArgumentException` which the interface lists as what it throws.